src/Symfony/Component/Messenger/Stamp/UniqueStamp.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Stamp;
+
+/**
+ * Added by the user to signify this message is meant to be only one of this unique message inside
+ * of the message queue. This can be used to prevent enqueueing the same message awaiting to be processed.
+ *
+ * @author RJ Garcia <ragboyjr@icloud.com>
+ */
+final class UniqueStamp implements StampInterface
+{
+    private $id;
+
+    public function __construct(?string $id = null)
```

Comment:
What sense add this stamp without providing of id?

Reply:
The transport could generate a unique id from like a hash of the message body. SQS for example supports both client supplied ids or allowing sqs to dedup it on their end.

Reply:
But this is definitely something I should make clear in the documentation.

Reply:
Is it needed to enforce string or null? Could we allow int?

Reply:
Hmm, i'm not sure, SQS uses strings up to 100 chars, and if the id is null, it's expected to be a hash of the serialized body which is also a string, and ints are easily typecast-able to strings.

If I could typehint `?int|string`, then I think it'd be a no brainer, but that's not available till php 8. I definitely prefer getting to use a php type decl vs a docblock, but if the consensus of the core team is that we should allow both, then i'm cool with that.

Reply:
I am definitely 👍🏻 for a typehint but instead of just having a docblock you could check the type inside the method